pwa/src/pages/entities/[id].tsx

```diff
@@ -70,7 +70,7 @@ const IndexPage = (props) => {
                 aria-labelledby="data-tab"
               >
                 <br/>
-                <DataTable {...{id}}/>
+                <DataTable id={props.params.id}/>
```

Comment:
Hier was een refactor geschreven naar `{...{id}}` zodat `id` nooit meer `"new"` zou zijn. Wat logischer is, aangezien "new" geen `id` is. Op regel `11` van dit bestand wordt de volgende variabele geset: 

```
const id: string = props.params.id === "new" ? null : props.params.id
```

Hierdoor kunnen we in de diepere bestanden een simpelere check doen waarbij we checken óf `id` bestaat i.p.v. of die "new" is of iets anders. Als dit iets breekt dieper in de applicatie dan zou mijn voorkeur gaan naar dieper in de applicatie de checks voor `id === "new"` weg te halen en om te schrijven naar de check of `id` bestaat. 

Voorbeeld in `applicationForm.tsx`, hier wordt zo'n check gedaan:

```
id && handleSetApplications()
```

Er zit hier een diepere gedachtegang achter waarom dit belangrijk is. Componenten moeten zo min mogelijk context van buitenaf moeten afhandelen. We willen ze zoveel mogelijk encapsulated houden. Wanneer we in een component moeten checken of `id === "new"` dan zijn we die logica op de verkeerde plek aan het plaatsen (stel je voor dat in een ander scenario ook `"old"` mee willen gaan sturen, dan gaat dit component kapot).

_Out of scope, maar misschien wel interessant: wanneer we dit nog één stap verder trekken zou de check voor `id` helemaal niet plaats moeten vinden in een component maar zouden we een component moeten hebben die wél een `id` verwacht en een component die niet een `id` verwacht en ze volledig van elkaar lostrekken. Dat lijkt initieel onhandig, omdat je dan code moet dupliceren. Maar in grotere applicaties is het veel beter te beheren._